### PR TITLE
Make request header blacklist case insensitive

### PR DESCRIPTION
**Pull request by thewilkybarkid**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |
| License       | Apache License 2.0

### CODE REVIEW

**Comment by csarrazi** [comment]
Actually, this is a BC break, as it changes the file naming strategy.
This means that you will need to regenerate your mocks once this patch is applied.

**Comment by thewilkybarkid** [comment]
Ok, I've changed it so it tries normalising the request headers first, and falls back to not normalising.

**Comment by csarrazi** [comment]
We're nearly there!

Could you add a test to check that both the new and the old naming strategies return the expected paths? 

Thanks! 😃

File: tests/Cache/MockStorageAdapterTest.php [hunk 1]
```diff
@@ -66,8 +66,9 @@ public function testFetch()
     public function testSave()
     {
         $request = $this->getRequestMock();
-        $mockStorage = new $this->class($this->tmpDir, [], ['X-Foo']);
```
Comment (csarrazi) [inline tests/Cache/MockStorageAdapterTest.php]: This test should not be altered.

On the other hand, we should create tests which test specifically the naming strategies.

**Comment by csarrazi** [comment]
I'm closing this PR as the logic has been moved to a third-party library. @thewilkybarkid Want to make a PR against the new project? (https://github.com/csarrazi/guzzle-cache-middleware) :)

**Pull request closed without merging**